xbmc/windowing/gbm/drm/DRMAtomic.cpp

```diff
@@ -60,17 +46,13 @@ uint32_t GetScalingFactor(uint32_t srcWidth,
 
 bool CDRMAtomic::SetScalingFilter(struct CDRMObject* object, const char* name, const char* type)
 {
-  uint32_t filter_type = GetScalingFilterType(type);
-  if (object->SupportsPropertyAndValue(name, filter_type))
+  uint64_t filter_type = GetScalingFilterType(type);
+  if (m_gui_plane->GetPropertyValue(name, type, filter_type))
   {
     if (AddProperty(object, name, filter_type))
     {
       uint32_t mar_scale_factor =
           GetScalingFactor(m_width, m_height, m_mode->hdisplay, m_mode->vdisplay);
-      uint32_t diff_w = m_mode->hdisplay - (mar_scale_factor * m_width);
-      uint32_t diff_h = m_mode->vdisplay - (mar_scale_factor * m_height);
-      AddProperty(object, "CRTC_X", (diff_w / 2));
-      AddProperty(object, "CRTC_Y", (diff_h / 2));
       AddProperty(object, "CRTC_W", (mar_scale_factor * m_width));
       AddProperty(object, "CRTC_H", (mar_scale_factor * m_height));
```

Comment:
As I said, setting the CRTC X, Y, W, H to center the integer-scaled image is fine. You just need to prepare for the fact that the KMS driver may not support that, particularly if the KMS plane happens to be the primary plane.

If that does not work, the it is very likely that you cannot have `"CRTC_W", (mar_scale_factor * m_width)` or `"CRTC_H", (mar_scale_factor * m_height)` either. How do you handle it when the KMS driver refuses?

It looks like `CDRMAtomic::DrmAtomicCommit()` has literally no fallback plan. It attempts a test-only commit, yes, but if that fails, you are supposed to try something else until it succeeds. Right now it looks like the user would be left with a black or frozen screen.

So you would try with the centered integer-scaled image first. If that fails, then must try again without integer scaling.

Reply:
Please, explain how this review comment has been addressed, or alternatively why it does not need addressing?